source/controlTypes/role.py

```diff
@@ -0,0 +1,531 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2007-2021 NV Access Limited, Babbage B.V.
+
+from enum import IntEnum
+from typing import Dict, Set
+
+from utils.displayString import DisplayStringEnumMixin, DisplayStringEnumMixinMeta
+
+
+class Role(DisplayStringEnumMixin, IntEnum, metaclass=DisplayStringEnumMixinMeta):
+	@property
+	def _displayStringLabels(self):
+		return _roleLabels
+
+	@property
+	def defaultValue(self):
+		return self.UNKNOWN
+
+	UNKNOWN = 0
+	WINDOW = 1
+	TITLEBAR = 2
+	PANE = 3
```

Comment:
Thoughts on replacing this with the following?

```suggestion
	UNKNOWN = 0
	WINDOW = auto()
	TITLEBAR = auto()
	PANE = auto()
	...
```

Reply:
I don't think this is a good idea as It'll break backwards compatibility for add-ons who're using integer values of these roles rather than constants. I agree this is a bad practice but I've seen this in the wild (the only example which I was able to recall quickly was @josephsl controlUsageAssistant) but there are almost certainly more. The reason this is not going to work is the fact that not all these roles are numbered sequentially for example `FOOTNOTE` is 67 and `GLASSPANE` not 68 but 69. It should be possible to implement `_generate_next_value_` for the enum and do something clever inside but IMO it is not worth the trouble.

Reply:
`WINDOW = auto()` is the equivalent of `WINDOW = 1`, however it makes it less obvious what the value is if you are using integers in a different system.

Reply:
Yes, this is backwards compatible:
```python
>>> from enum import IntEnum
>>> class blah(IntEnum):
...     first = 0
...     second = enum.auto()
...
>>> blah.first == 0
True
>>> blah.second == 1
True
>>>
```

However, where an `enum` is matching up with some other definition, EG defined in Windows, I think it is better to be explicit.

Reply:
It is backward compatible if and only if we would use `auto` only up to the `FOOTNOTE` and then  start using integers explicitly again, since as I've explained above numbering is no longer sequential from this point onward. I don't see any advantage to doing this though.